recipe-app/src/pages/recipesList/components/EditRecipeModal/EditIngredientsField.tsx

```diff
@@ -0,0 +1,55 @@
+import { Ingredient } from "../../types";
+import { useField } from "react-final-form";
+import { Chip, Grid, Typography } from "@mui/material";
+import { useState } from "react";
+
+export const EditIngredientsField = () => {
+  const ingredientsField = useField<Ingredient[]>("ingredients");
+
+  const ingredients = ingredientsField.input.value || [];
+
+  const handleDelete = (index: number) => {
+    ingredientsField.input.onChange([
+      ...ingredients.slice(0, index),
+      ...ingredients.slice(index + 1),
+    ]);
```

Comment:
This is a bit strange too, normally you would update a state here. Something like this?
```
const [ingredients, setIngredients] = useState<Ingredient[]>([])
const handleDelete = (index: number) => {
  setIngredients(currentIngredients => currentIngredients.splice(index, 1))
}
const handleAdd = () = > {
  setIngredients(currentIngredients => [...currentIngredients, { name: newIngredientName }]
}
```

Reply:
Yea this  `field.input.onChange` is due to the [react-final-form ](https://final-form.org/react/) library as well. One of the benefit of this library is also that we can manipulate the state here inside the field component. Otherwise we would have to keep this useState logic not here, but in the parent form component where we need access to the ingredients to make the final submit call. 

Also, If you meant why i used the slice methods and not `currentIngredients.splice(index, 1))`, because, splice does an in-place update to the array and returns the deleted items, so we cannot use that here.

Reply:
Oh you're right, splice wouldn't work in mine either then